axon/src/main/java/com/baeldung/axon/gui/OrderRestEndpoint.java

```diff
@@ -91,4 +97,20 @@ public CompletableFuture<Void> shipOrder(@PathVariable("order-id") String orderI
     public CompletableFuture<List<Order>> findAllOrders() {
         return queryGateway.query(new FindAllOrderedProductsQuery(), ResponseTypes.multipleInstancesOf(Order.class));
     }
+
+    @GetMapping("/total-shipped/{product-id}")
+    public Integer totalShipped(@PathVariable("product-id") String productId) {
+        return queryGateway.scatterGather(new TotalProductsShippedQuery(productId), ResponseTypes.instanceOf(Integer.class), 10L, TimeUnit.SECONDS)
+                .reduce(0, Integer::sum);
+    }
+
+    @GetMapping(path = "/order-updates/{order-id}", produces = MediaType.TEXT_EVENT_STREAM_VALUE)
+    public Flux<Order> orderUpdates(@PathVariable("order-id") String orderId) {
```

Comment:
Perhaps it's nicer to work with a dedicated `OrderResponse` kind of object, instead of returning the `Order` model. 
To not expose the model in essence.
What do you think?

Reply:
Yes, I think some additional decoupling would be nice and good to add.